EIPS/eip-5089.md

```diff
@@ -0,0 +1,393 @@
+schemes.
+
+The primary examples include yield tokenization platforms which strip future yield leaving a principal
+token behind, as well as fixed-rate money-markets which utilize principal tokens as a medium
+to lend/borrow.
+
+This inconsistency in implementation makes integration difficult at the application layer as well as
+wallet layer which are key catalysts for the space's growth. 
+Developers are currently expected to implement individual adapters for each principal token, as well as adapters for
+their pool contracts, and many times adapters for their custodial contracts as well, wasting significant developer resources. 
+
+## Specification
+
+All Principal Tokens (PTs) MUST implement [EIP-20](./eip-20.md) to represent ownership of future underlying redemption.
+If a PT is to be non-transferrable, it MAY revert on calls to `transfer` or `transferFrom`.
+The [EIP-20](./eip-20.md) operations `balanceOf`, `transfer`, `totalSupply`, etc. operate on the Principal Token balance.
+
+All Principal Tokens MUST implement [EIP-20](./eip-20.md)'s optional metadata extensions.
+The `name` and `symbol` functions SHOULD reflect the underlying token's `name` and `symbol` in some way, as well as the origination protocol, and in the case of yield tokenization protocols, the origination money-market.
+
+All Principal Tokens MAY implement [EIP-2612](./eip-2612.md) to improve the UX of approving PTs on various integrations.
+
+### Definitions:
+
+- underlying: The token that Principal Tokens are redeemable for at maturity.
+  Has units defined by the corresponding [EIP-20](./eip-20.md) contract.
+- maturity: The timestamp (unix) at which a Principal Token matures. Principal Tokens become redeemable for underlying at or after this timestamp.
+- fee: An amount of underlying or Principal Token charged to the user by the Principal Token. Fees can exist on redemption or post-maturity yield.
+- slippage: Any difference between advertised redemption value and economic realities of PT redemption, which is not accounted by fees.
+
+### Methods
+
+#### `underlying`
+
+The address of the underlying token used by the Principal Token for accounting, and redeeming.
+
+MUST be an EIP-20 token contract.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: underlying
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: underlyingAddress
+      type: address
+```
+
+#### `maturity`
+
+The unix timestamp (uint256) at or after which Principal Tokens can be redeemed for their underlying deposit.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturity
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: timestamp
+      type: uint256
+```
+
+#### `convertToUnderlying`
+
+The amount of underlying that would be exchanged for the amount of PTs provided, in an ideal scenario where all the conditions are met.
+
+Before maturity, the amount of underlying returned is as if the PTs would be at maturity.
```

Comment:
Shouldn't the amount of underlying returned at maturity generally be a static amount of underlying unless there's a shortfall? 

Something to clarify for my own understanding – is that true? Is this standard for zero-coupon-like tokens with fixed payouts at maturity? (vs a floating payouts). If that's a correct understanding, then perhaps the `balance` of the token could carry the meaning of "intended payout in underlying at maturity".

What do you think?

Reply:
You get the intended payout at maturity by calling `convertToUnderlying` before maturity.

And as before, I don't see a technical reason to state what the payout should be at or after maturity.

Reply:
I still don't understand, do you want to enforce a redemption rate on the standard, or provide a purpose-built function to let users know what that rate is?

Reply:
The former. Maybe i'm not explaining myself well, i'll formulate my thoughts and bring this discussion point up again in the thread or the channel

Reply:
I don't see the point of enforcing a redemption rate, but please let's continue this conversation in the ethereum magicians thread.